src/main/java/net/fabricmc/fabric/mixin/events/tick/MixinPlayerEntity.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2016, 2017, 2018 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.events.tick;
+
+import net.fabricmc.fabric.api.event.player.PlayerTickCallback;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.world.World;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+@Mixin(PlayerEntity.class)
+public abstract class MixinPlayerEntity extends LivingEntity {
+	protected MixinPlayerEntity(EntityType<? extends LivingEntity> type, World world) {
+		super(type, world);
+	}
+
+	@Inject(method = "update", at = @At("TAIL"))
+	public void tickPlayer(CallbackInfo ci) {
+		PlayerEntity player = (PlayerEntity)(Object)this;
```

Comment:
Creating a local variable here each tick is redundant, this can just be
```java
PlayerTickCallback.EVENT.invoker().tick((PlayerEntity) (Object) this);
```